In [3]:
import pandas as pd

##Условие 1: 1 задача
- Скачать файл в уроке
- Считать данные с помощью pandas
- Вывести на экран первые 5 строк
- Посмотреть на описание признаков и на их содержание

In [4]:
df = pd.read_csv('../../data/kc-house-data.csv')

In [5]:
df.head(5)

id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0      5650     1.0           0     0  ...      7        1180              0   
1      7242     2.0           0     0  ...      7        2170            400   
2     10000     1.0           0     0  ...      6         770              0   
3      5000     1.0           0     0  ...      7        1050            910   
4      8080     1.0           0     0  ...      8        1680              0   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0      1955             0    98178  47.5112 -122.257           1340   
1      1951          1991    98125  47.7210 -122.319           1690   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   

   sqft_lot15  
0        5650  
1        7639  
2        8062  
3        5000  
4        7503  

[5 rows x 21 columns]


##Условие 2: 2 задача
Проведите первичный анализ данных
- Изучите типы данных
- Найдите количество пропущенных ячеек в данных
- Посчитайте основные статистики по всем признакам и поизучайте их
- Пишите выводы

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21613 entries, 0 to 21612
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             21613 non-null  int64  
 1   date           21613 non-null  object 
 2   price          21613 non-null  float64
 3   bedrooms       21613 non-null  int64  
 4   bathrooms      21613 non-null  float64
 5   sqft_living    21613 non-null  int64  
 6   sqft_lot       21613 non-null  int64  
 7   floors         21613 non-null  float64
 8   waterfront     21613 non-null  int64  
 9   view           21613 non-null  int64  
 10  condition      21613 non-null  int64  
 11  grade          21613 non-null  int64  
 12  sqft_above     21613 non-null  int64  
 13  sqft_basement  21613 non-null  int64  
 14  yr_built       21613 non-null  int64  
 15  yr_renovated   21613 non-null  int64  
 16  zipcode        21613 non-null  int64  
 17  lat            21613 non-null  float64
 18  long  

In [7]:
df.describe()

id         price      bedrooms     bathrooms   sqft_living  \
count  2.161300e+04  2.161300e+04  21613.000000  21613.000000  21613.000000   
mean   4.580302e+09  5.400881e+05      3.370842      2.114757   2079.899736   
std    2.876566e+09  3.671272e+05      0.930062      0.770163    918.440897   
min    1.000102e+06  7.500000e+04      0.000000      0.000000    290.000000   
25%    2.123049e+09  3.219500e+05      3.000000      1.750000   1427.000000   
50%    3.904930e+09  4.500000e+05      3.000000      2.250000   1910.000000   
75%    7.308900e+09  6.450000e+05      4.000000      2.500000   2550.000000   
max    9.900000e+09  7.700000e+06     33.000000      8.000000  13540.000000   

           sqft_lot        floors    waterfront          view     condition  \
count  2.161300e+04  21613.000000  21613.000000  21613.000000  21613.000000   
mean   1.510697e+04      1.494309      0.007542      0.234303      3.409430   
std    4.142051e+04      0.539989      0.086517      0.766318      0.650743   
min    5.200000e+02      1.000000      0.000000      0.000000      1.000000   
25%    5.040000e+03      1.000000      0.000000      0.000000      3.000000   
50%    7.618000e+03      1.500000      0.000000      0.000000      3.000000   
75%    1.068800e+04      2.000000      0.000000      0.000000      4.000000   
max    1.651359e+06      3.500000      1.000000      4.000000      5.000000   

              grade    sqft_above  sqft_basement      yr_built  yr_renovated  \
count  21613.000000  21613.000000   21613.000000  21613.000000  21613.000000   
mean       7.656873   1788.390691     291.509045   1971.005136     84.402258   
std        1.175459    828.090978     442.575043     29.373411    401.679240   
min        1.000000    290.000000       0.000000   1900.000000      0.000000   
25%        7.000000   1190.000000       0.000000   1951.000000      0.000000   
50%        7.000000   1560.000000       0.000000   1975.000000      0.000000   
75%        8.000000   2210.000000     560.000000   1997.000000      0.000000   
max       13.000000   9410.000000    4820.000000   2015.000000   2015.000000   

            zipcode           lat          long  sqft_living15     sqft_lot15  
count  21613.000000  21613.000000  21613.000000   21613.000000   21613.000000  
mean   98077.939805     47.560053   -122.213896    1986.552492   12768.455652  
std       53.505026      0.138564      0.140828     685.391304   27304.179631  
min    98001.000000     47.155900   -122.519000     399.000000     651.000000  
25%    98033.000000     47.471000   -122.328000    1490.000000    5100.000000  
50%    98065.000000     47.571800   -122.230000    1840.000000    7620.000000  
75%    98118.000000     47.678000   -122.125000    2360.000000   10083.000000  
max    98199.000000     47.777600   -121.315000    6210.000000  871200.000000

In [14]:
df['bathrooms'].value_counts()

2.50    5380
1.00    3852
1.75    3048
2.25    2047
2.00    1930
1.50    1446
2.75    1185
3.00     753
3.50     731
3.25     589
3.75     155
4.00     136
4.50     100
4.25      79
0.75      72
4.75      23
5.00      21
5.25      13
0.00      10
5.50      10
1.25       9
6.00       6
0.50       4
5.75       4
6.75       2
8.00       2
6.25       2
6.50       2
7.50       1
7.75       1
Name: bathrooms, dtype: int64

In [16]:
df['bedrooms'].value_counts()

3     9824
4     6882
2     2760
5     1601
6      272
1      199
7       38
0       13
8       13
9        6
10       3
11       1
33       1
Name: bedrooms, dtype: int64

- Пропущеных ячеек нет.
- Основная часть домов имеет 3 или 4 спальни, при этом количество ванных комнат от 1 до 3.
- Половина домов одноэтажные, другая половина домов в основном двухэтажные.
- Большая часть домов не имеет подвала.


##Условие 3: 3 задача

Ответьте на несколько вопросов
- В каком диапазоне изменяются стоимости недвижимости?
- Какую долю в среднем занимают жилая площадь от всей площади по всем домам?
- Как много домов с разными этажами в данных?
- Насколько хорошие состояния у домов в данных?
- Найдите года, когда построили первый дом, когда построили последний дом в данных?

In [18]:
(df['sqft_living'].mean() / df['sqft_lot'].mean()) * 100

13.76781757959227

In [19]:
len(df[df['floors'] > 1])

10933

In [20]:
df['condition'].value_counts()

3    14031
4     5679
5     1701
2      172
1       30
Name: condition, dtype: int64

- стоимости недвижимости изменяются в диапазоне от 7.500000e+04	до 7.700000e+06.
- жилая площадь в среднем занимает 14% от всей площади по всем домам.
- домов имеющих более одного этажа - 10933
- основная часть домов имеет оценку состояния 3 из 5.
- первый дом построили в 1900 году,  последний дом построили в 2015 году.


##Условие 4: 4 задача

Ответьте на несколько вопросов

- Сколько в среднем стоят дома, у которых 2 спальни?
- Какая в среднем общая площадь домов, у которых стоимость больше 600 000?
- Как много домов коснулся ремонт?
- Насколько в среднем стоимость домов с оценкой grade домов выше 10 отличается от стоимости домов с оценкой grade меньше 4?

In [27]:
df[df['bedrooms'] == 2]['price'].mean()

401372.681884058

In [28]:
df[df['price'] > 600000]['sqft_lot'].mean()

20442.524776214832

In [29]:
df[df['yr_renovated'] > 0]['sqft_lot'].count()

914

In [44]:
df[df['grade'] > 10]['price'].mean() - df[df['grade'] < 4]['price'].mean()

1488885.1175298805

- дома с двумя спальнями в среднем стоят 401372.68
- общая площадь домов, у которых стоимость больше 600 000 в среднем составляет 20442.52
- 914 домов подверглись капитальному ремонту
- стоимость домов с оценкой grade домов выше 10 отличается от стоимости домов с оценкой grade меньше 4 в среднем  на 1488885.1175298805.

##Условие 5: 5 задача

Ответьте на несколько вопросов
-  Выберите дом клиенту. Клиент хочет дом с видом на набережную, как минимум с тремя ванными и с подвалом. Сколько вариантов есть у клиента?
-  Выберите дом клиенту. Клиент хочет дом либо с очень красивым видом из окна, либо с видом на набережную, в очень хорошем состоянии и год постройки не меньше 1980 года. В какой ценовом диапазоне будут дома?
-  Выберите дом клиенту. Клиент хочет дом без подвала, с двумя этажами, стоимостью до 150000. Какая оценка по состоянию у таких домов в среднем?

In [36]:
len(df[(df.waterfront == 1) & (df.bathrooms >= 3) & (df.sqft_basement > 0)])

41

In [37]:
df.view.value_counts()

0    19489
2      963
3      510
1      332
4      319
Name: view, dtype: int64

In [42]:
df[((df.waterfront == 1) | (df.view == 4)) & (df.condition == 5) & (df.yr_built >= 1980)].describe()

id         price  bedrooms  bathrooms  sqft_living  \
count  5.000000e+00  5.000000e+00  5.000000   5.000000     5.000000   
mean   6.067580e+09  1.789320e+06  2.600000   2.850000  3686.000000   
std    2.199073e+09  6.958016e+05  0.894427   0.782624   845.357912   
min    3.585901e+09  1.295000e+06  2.000000   2.500000  2910.000000   
25%    3.761100e+09  1.411600e+06  2.000000   2.500000  3180.000000   
50%    7.278101e+09  1.505000e+06  2.000000   2.500000  3180.000000   
75%    7.856400e+09  1.735000e+06  3.000000   2.500000  4310.000000   
max    7.856400e+09  3.000000e+06  4.000000   4.250000  4850.000000   

           sqft_lot    floors  waterfront  view  condition      grade  \
count      5.000000  5.000000    5.000000   5.0        5.0   5.000000   
mean   16557.400000  1.900000    0.400000   4.0        5.0   9.800000   
std     9605.789936  0.223607    0.547723   0.0        0.0   0.447214   
min     9400.000000  1.500000    0.000000   4.0        5.0   9.000000   
25%     9400.000000  2.000000    0.000000   4.0        5.0  10.000000   
50%    12445.000000  2.000000    0.000000   4.0        5.0  10.000000   
75%    19449.000000  2.000000    1.000000   4.0        5.0  10.000000   
max    32093.000000  2.000000    1.000000   4.0        5.0  10.000000   

        sqft_above  sqft_basement    yr_built  yr_renovated       zipcode  \
count     5.000000        5.00000     5.00000           5.0      5.000000   
mean   2704.000000      982.00000  1985.20000           0.0  98080.000000   
std     698.627225      502.26487     2.48998           0.0     89.283257   
min    1940.000000      570.00000  1982.00000           0.0  98006.000000   
25%    2510.000000      570.00000  1985.00000           0.0  98006.000000   
50%    2610.000000      970.00000  1985.00000           0.0  98034.000000   
75%    2610.000000     1000.00000  1985.00000           0.0  98177.000000   
max    3850.000000     1800.00000  1989.00000           0.0  98177.000000   

             lat        long  sqft_living15   sqft_lot15  
count   5.000000    5.000000       5.000000      5.00000  
mean   47.671960 -122.266600    3444.000000  16221.60000  
std     0.104322    0.115038     538.451483   8151.88891  
min    47.561700 -122.393000    2540.000000   9450.00000  
25%    47.561700 -122.380000    3350.000000   9450.00000  
50%    47.701100 -122.244000    3760.000000  12210.00000  
75%    47.762400 -122.158000    3760.000000  23598.00000  
max    47.772900 -122.158000    3810.000000  26400.00000

In [43]:
df[(df.price > 150000) & (df.floors == 2) & (df.sqft_basement == 0)].condition.mean()

3.162087912087912

-  Если клиент хочет дом с видом на набережную, как минимум с тремя ванными и с подвалом, у клиента есть 41 вариант.
-  Если клиент хочет дом либо с очень красивым видом из окна, либо с видом на набережную, в очень хорошем состоянии и год постройки не меньше 1980 года, дома будут в ценовом диапазоне от 1295000 до 3000000
-  Если клиент хочет дом без подвала, с двумя этажами, стоимостью до 150000, средняя оценка по состоянию у таких домов равна 3.162